v3/pkg/application/messageprocessor_dialog.go

```diff
@@ -5,19 +5,16 @@ import (
 	"fmt"
 	"net/http"
 	"runtime"
-	"strconv"
 )
 
 func (m *MessageProcessor) dialogErrorCallback(window *WebviewWindow, message string, dialogID *string, err error) {
 	errorMsg := fmt.Sprintf(message, err)
 	m.Error(errorMsg)
-	msg := "_wails.dialogErrorCallback('" + *dialogID + "', " + strconv.Quote(errorMsg) + ");"
-	window.ExecJS(msg)
+	window.DialogError(dialogID, errorMsg)
 }
 
 func (m *MessageProcessor) dialogCallback(window *WebviewWindow, dialogID *string, result string, isJSON bool) {
-	msg := fmt.Sprintf("_wails.dialogCallback('%s', %s, %v);", *dialogID, strconv.Quote(result), isJSON)
```

Comment:
If we remove `isJSON` here, do multiple file select dialogs still work?

Reply:
Perhaps this method could always just decode the payload? I believe a string is a valid json payload. https://github.com/wailsapp/wails/blob/master/v3/internal/runtime/desktop/dialogs.js#L35-L45

Reply:
Let me check the `file dialog` stuff.  I have to be honest in that I didn't test that since I haven't gotten that reimplemented in Linux yet.  That is the next thing on my todo list after the menus.

Reply:
I'll check it on the `mac`.

Reply:
https://github.com/wailsapp/wails/pull/2587/files#diff-5f2cc1ceaf1ea4e502ed0aca10ee2858b1f6978b278b2200e27c5d8746e9a8deL66 uses `false`

Reply:
Just tested multiple dialogs on Mac.  Works fine without any other changes so far.